packages/core/src/storage/compare.ts

```diff
@@ -60,6 +60,42 @@ export interface LayoutChange {
   bytes?: Record<'from' | 'to', string>;
 }
 
+// StorageItem with slot and offset info. Needed for overlap detection
+type StorageItemFull = StorageItem & { slot: string; offset: number };
+
+function isFullInfo(entry: StorageField): entry is StorageItemFull {
+  return entry.slot != undefined && entry.offset != undefined;
+}
+
+function storageItemBegin(entry: StorageItemFull): number {
+  return Number(entry.slot) * 32 + entry.offset;
+}
+
+function storageItemEnd(entry: StorageItemFull): number {
+  return storageItemBegin(entry) + roundUpBytes(Number(entry.type.item.numberOfBytes));
```

Comment:
I'm not sure about that. `storageItemBegin` is not necessary "aligned".

We could have an `entry` with slot: 10, offset: 4, number of bytes: 4.
The beginning is at `10*32+4=324`  and the end is at `10*32+4+4=328`. The entry is in slot #10, which goes from 320 to 351

Yet the roundUp will say that is starts at 324, and that is ends at 356, which is wrong.

I'm not sure where the `roundUpBytes` is actually needed, but I think we should have 
```
function storageItemEnd(entry: StorageItemFull): number {
  return storageItemBegin(entry) + Number(entry.type.item.numberOfBytes);
}
```
and possibly doing `roundUpBytes(storageItemEnd(entry))` in some places.